Lib/test/test_asyncio/test_queues.py

```diff
@@ -522,5 +522,221 @@ class PriorityQueueJoinTests(_QueueJoinTestMixin, unittest.IsolatedAsyncioTestCa
     q_class = asyncio.PriorityQueue
 
 
+class _QueueShutdownTestMixin:
+    q_class = None
+
+    @staticmethod
+    async def _ensure_started(task):
+        # Explicitly start (if not already) task
+
+        async def swallow(x):
+            try:
+                return await x
+            except Exception:
+                pass
+
+        try:
+            await asyncio.wait_for(asyncio.shield(swallow(task)), 0.01)
+        except TimeoutError:
+            pass
+
+    def assertRaisesShutdown(self, msg="Didn't appear to shut-down queue"):
+        return self.assertRaises(asyncio.QueueShutDown, msg=msg)
+
+    async def test_format(self):
+        q = self.q_class()
+        q._is_shutdown = True
+        self.assertEqual(q._format(), 'maxsize=0 shutdown')
+
+    async def test_shutdown_empty(self):
+        # Test shutting down an empty queue
+
+        # Setup empty queue, and join() and get() tasks
+        q = self.q_class()
+        loop = asyncio.get_running_loop()
+        join_task = loop.create_task(q.join())
+        get_task = loop.create_task(q.get())
+        await self._ensure_started(get_task)  # want pending before shutdown
```

Comment:
This will start all the tasks. In particular, `q.join()` will start and immediately complete (since the queue is empty at this point).